pkg/operator/clustermembercontroller/clustermembercontroller.go

```diff
@@ -101,21 +118,52 @@ func (c *ClusterMemberController) reconcileMembers(ctx context.Context, recorder
 	case podToAdd == nil:
 		// no more work left to do
 		return nil
-	default:
-		recorder.Eventf("FoundPodToScale", "found pod to add to etcd membership: %v", podToAdd.Name)
 	}
 
 	etcdHost, err := c.getEtcdPeerHostToScale(podToAdd)
 	if err != nil {
 		return fmt.Errorf("could not get etcd peer host :%w", err)
 	}
+
+	if machineForHostPendingDeletion, err := c.isMachinePendingDeletionFor(etcdHost); err != nil || machineForHostPendingDeletion {
```

Comment:
my personal preference is to have an if/else-if with a return err and a return nil, but I'm willing to defer to @hasbro17